homeassistant/components/edl21/sensor.py

```diff
@@ -32,8 +32,39 @@
 class EDL21:
     """EDL21 handles telegrams sent by a compatible smart meter."""
 
+    # OBIS format: A-B:C.D.E*F
+    _OBIS_NAMES = {
+        # A=1: Electricity
+        # C=0: General purpose objects
+        "1-0:0.0.9*255": "Electricity ID",
+        # C=1: Active power +
+        # D=8: Time integral 1
+        # E=0: Total
+        "1-0:1.8.0*255": "Positive active energy total",
+        # E=1: Rate 1
+        "1-0:1.8.1*255": "Positive active energy in tariff T1",
+        # E=2: Rate 2
+        "1-0:1.8.2*255": "Positive active energy in tariff T2",
+        # D=17: Time integral 7
+        # E=0: Total
+        "1-0:1.17.0*255": "Last signed positive active energy total",
+        # C=15: Active power absolute
+        # D=7: Instantaneous value
+        # E=0: Total
+        "1-0:15.7.0*255": "Absolute active instantaneous power",
+        # C=16: Active power sum
+        # D=7: Instantaneous value
+        # E=0: Total
+        "1-0:16.7.0*255": "Sum active instantaneous power",
+    }
+
     def __init__(self, hass, config, async_add_entities) -> None:
         """Initialize an EDL21 object."""
+        self._blacklist = {
```

Comment:
Why have the blacklist as an instance variable and not a class variable like `_OBIS_NAMES`? Shouldn't we just warn once for each type across every entity?

Reply:
Done.